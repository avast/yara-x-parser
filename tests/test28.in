//Rule comment
rule test
{
	//Rule block comment
	strings:
		$a = "foo"
		$b = "bar"
	condition:
		$a and ($b or $a) and not $b
}
