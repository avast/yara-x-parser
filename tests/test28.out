SOURCE_FILE@0..130
  RULE@0..129
    COMMENT@0..14 "//Rule comment"
    WHITESPACE@14..15 "\n"
    RULE_KW@15..19 "rule"
    WHITESPACE@19..20 " "
    IDENTIFIER@20..24 "test"
    WHITESPACE@24..25 "\n"
    BLOCK_EXPR@25..129
      L_BRACE@25..26 "{"
      WHITESPACE@26..28 "\n\t"
      STRINGS@28..84
        COMMENT@28..48 "//Rule block comment"
        WHITESPACE@48..50 "\n\t"
        STRINGS_KW@50..57 "strings"
        COLON@57..58 ":"
        WHITESPACE@58..61 "\n\t\t"
        VARIABLE_STMT@61..71
          VARIABLE@61..63 "$a"
          WHITESPACE@63..64 " "
          ASSIGN@64..65 "="
          WHITESPACE@65..66 " "
          PATTERN@66..71
            STRING_LIT@66..71 "\"foo\""
        WHITESPACE@71..74 "\n\t\t"
        VARIABLE_STMT@74..84
          VARIABLE@74..76 "$b"
          WHITESPACE@76..77 " "
          ASSIGN@77..78 "="
          WHITESPACE@78..79 " "
          PATTERN@79..84
            STRING_LIT@79..84 "\"bar\""
      WHITESPACE@84..86 "\n\t"
      CONDITION@86..127
        CONDITION_KW@86..95 "condition"
        COLON@95..96 ":"
        WHITESPACE@96..99 "\n\t\t"
        EXPRESSION_STMT@99..127
          BOOLEAN_EXPR@99..127
            BOOLEAN_EXPR@99..116
              BOOLEAN_TERM@99..101
                VARIABLE@99..101 "$a"
              WHITESPACE@101..102 " "
              AND_KW@102..105 "and"
              WHITESPACE@105..106 " "
              BOOLEAN_EXPR@106..116
                BOOLEAN_TERM@106..116
                  L_PAREN@106..107 "("
                  BOOLEAN_EXPR@107..115
                    BOOLEAN_TERM@107..109
                      VARIABLE@107..109 "$b"
                    WHITESPACE@109..110 " "
                    OR_KW@110..112 "or"
                    WHITESPACE@112..113 " "
                    BOOLEAN_EXPR@113..115
                      BOOLEAN_TERM@113..115
                        VARIABLE@113..115 "$a"
                  R_PAREN@115..116 ")"
            WHITESPACE@116..117 " "
            AND_KW@117..120 "and"
            WHITESPACE@120..121 " "
            BOOLEAN_EXPR@121..127
              BOOLEAN_TERM@121..127
                NOT_KW@121..124 "not"
                WHITESPACE@124..125 " "
                BOOLEAN_TERM@125..127
                  VARIABLE@125..127 "$b"
      WHITESPACE@127..128 "\n"
      R_BRACE@128..129 "}"
  WHITESPACE@129..130 "\n"
